x/igp/types/msgs.go

```diff
@@ -0,0 +1,175 @@
+package types
+
+import (
+	"cosmossdk.io/math"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+var (
+	_ sdk.Msg = (*MsgCreateIgp)(nil)
+	_ sdk.Msg = (*MsgPayForGas)(nil)
+	_ sdk.Msg = (*MsgClaim)(nil)
+	_ sdk.Msg = (*MsgSetGasOracles)(nil)
+	_ sdk.Msg = (*MsgSetBeneficiary)(nil)
+	_ sdk.Msg = (*MsgSetRemoteGasData)(nil)
+	_ sdk.Msg = (*MsgSetDestinationGasOverhead)(nil)
+)
+
+// GetSigners implements sdk.Msg
+func (m MsgSetDestinationGasOverhead) GetSigners() []sdk.AccAddress {
+	signer, _ := sdk.AccAddressFromBech32(m.Sender)
+	return []sdk.AccAddress{signer}
+}
+
+// ValidateBasic implements sdk.Msg
+func (m MsgSetDestinationGasOverhead) ValidateBasic() error {
```

Comment:
Validation seems to be a common source of vulnerabilities in all blockchains and smart contracts. These look fine from a glance, but I'm curious if all edge cases were considered. 

It may be worth a future story and PR to double check or perhaps we wait for an audit (if that's happening).